src-java/base-topology/base-storm-topology/src/main/java/org/openkilda/wfm/AbstractBolt.java

```diff
@@ -127,16 +127,30 @@ protected void dispatch(Tuple input) throws Exception {
 
     protected void handleLifeCycleEvent(LifecycleEvent event) {
         if (Signal.START.equals(event.getSignal())) {
-            active = true;
             emit(ZkStreams.ZK.toString(), currentTuple, new Values(event, commandContext));
+            try {
+                activate();
+            } finally {
+                active = true;
```

Comment:
I would like to alter `active` value inside `activate()` / `deactivate()` methods, in this case successors can alter behaviour, if required.

Reply:
Good point, but there is no such case yet. For now, we want to always update this flag.